tests/fsm/run-test.sh

```diff
@@ -29,16 +31,46 @@ python3 generate.py -c $count $seed
 	for i in $( ls temp/*.ys | sed 's,[^0-9],,g; s,^0*\(.\),\1,g;' ); do
 		idx=$( printf "%05d" $i )
 		echo "temp/uut_${idx}.log: temp/uut_${idx}.ys temp/uut_${idx}.v"
-		echo "	@echo -n '[$i]'"
+		if test $verbose -gt 0; then
+			echo "	@echo -n 'Test fsm[$i] -> '"
+		fi
 		echo "	@../../yosys -ql temp/uut_${idx}.out temp/uut_${idx}.ys"
 		echo "	@mv temp/uut_${idx}.out temp/uut_${idx}.log"
-		echo "	@grep -q 'SAT proof finished' temp/uut_${idx}.log && echo -n K || echo -n T"
```

Comment:
This line succeeds no matter if "SAT proof finished" was found in temp/uut_${idx}.log. This is the intended behavior! It is okay to hit a timeout! The new code fails the test in case of a timeout. This is a bug.

If the proof fails, the call to `../../yosys` above already fails. Failing here on a timeout is a bug!

This is the same issue why I did not merge this in October, and I will not merge it now for the same reason. Fix this and I'll be happy to merge the PR.